packages/amplify-cli-core/src/types.ts

```diff
@@ -20,19 +20,209 @@ export type $TSContext = {
    * @deprecated
    */
   print: IContextPrint;
-  migrationInfo: $TSAny;
+  migrationInfo: MigrationInfo;
   projectHasMobileHubResources: boolean;
   prompt: $TSAny;
   exeInfo: $TSAny;
-  input: $TSAny;
-  parameters: $TSAny;
-  usageData: $TSAny;
-  runtime: $TSAny;
+  input: $CommandLineInput;
+  parameters: ContextParameters;
+  usageData: IUsageData;
+  runtime: Runtime;
   pluginPlatform: IPluginPlatform;
-  newUserInfo?: $TSAny;
+  newUserInfo?: string;
   filesystem: IContextFilesystem;
   template: IContextTemplate;
-  updatingAuth: $TSAny;
+  updatingAuth: any; // this seems to actually implement CognitoStackOptions
+};
+
+export interface MigrationInfo {
+  amplifyMeta: $TSMeta;
+  newVersion: string;
+  initVersion: string;
+  currentAmplifyMeta: $TSMeta;
+  projectConfig: ProjectConfig;
+  projectPath: string;
+  localEnvInfo: LocalEnvInfo;
+  localAwsInfo: LocalAwsInfo;
+  teamProviderInfo: TeamProviderInfo;
+  backendConfig: Record<string, unknown>;
+}
+export type TeamProviderEnvironment = {
+  categories: Record<string, unknown>;
+};
+export type TeamProviderInfo = {
+  [envName: string]: Record<string, unknown>;
+};
+export type LocalAwsInfo = {
+  NONE: unknown;
+};
+export type ProjectConfig<T extends string = ''> = {
+  frontend: T;
+  version: string;
+  providers: string[];
+  projectPath?: string;
+  defaultEditor?: string;
+  frontendHandler?: unknown;
+} & Record<T, string>;
+export interface LocalEnvInfo {
+  projectPath: string;
+  defaultEditor: string;
```

Comment:
defaultEditor and projectPath sound like candidates to go to `ProjectSettings` no ?